odoo/tools/sql.py

```diff
@@ -50,8 +50,19 @@ def table_kind(cr, tablename):
     cr.execute(query, (tablename,))
     return cr.fetchone()[0] if cr.rowcount else None
 
+SQL_TYPE_ORDER = {
+    'int4': 1,        # 4 bytes
+    'date': 2,        # 4 bytes
+    'numeric': 3,     # 4 bytes or variable
+    'varchar': 4,     # 4 bytes or variable
+    'text': 5,        # 4 bytes or variable
+    'bool': 6,        # 1 byte
```

Comment:
it would make more sense to me to put fixed size elements before variable size.

Reply:
I agree, this ordering doesn't make any sense to me.

Reply:
> it would make more sense to me to put fixed size elements before variable size.
> I agree, this ordering doesn't make any sense to me.

Size is not relevant, alignment is.

A bool has an alignment of 1, it can always fit at the end of the row in whatever space was left. That is not the case of variable-size columns (thereafter varlen), as noted above they are 4-aligned.

Since rows start with 4-aligned columns they are 4-aligned. In the worst case of putting bytes at the end the varlen ends on a mutliple of 4 and the byte triggers a padding of 3 bytes... but in that case it would have generated the exact same padding if it had been put before the varlen.

This means on average in 3/4 varlen lengths the boolean will fit in the varlen's slack, and in 1/4 cases it will not and will generate the exact same padding it would have otherwise.